app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3736,6 +3824,15 @@ private void notifyMetadataUpdateToListeners() {
         if (activityListener != null && currentMetadata != null) {
             activityListener.onMetadataUpdate(currentMetadata.getMetadata(), playQueue);
         }
+        if (currentMetadata != null && areSegmentsVisible
+                && segmentAdapter.setItems(currentMetadata.getMetadata())) {
+            final int adapterPosition = getNearestStreamSegmentPosition(
+                    simpleExoPlayer.getCurrentPosition());
+            segmentAdapter.selectSegmentAt(adapterPosition);
+            binding.itemsList.scrollToPosition(adapterPosition);
+        } else if (areSegmentsVisible) {
+            closeItemsList();
+        }
```

Comment:
Maybe wrap everything in a `if (areSegmentsVisible)`, to make it clearer:
```suggestion
        if (areSegmentsVisible) {
            if (currentMetadata != null
                    && segmentAdapter.setItems(currentMetadata.getMetadata())) {
                final int adapterPosition = getNearestStreamSegmentPosition(
                        simpleExoPlayer.getCurrentPosition());
                segmentAdapter.selectSegmentAt(adapterPosition);
                binding.itemsList.scrollToPosition(adapterPosition);
            } else {
                closeItemsList();
            }
        }
```

Reply:
Okay. I'll also move this code snippet - like the other one - to the upper method `onMetadataChanged` (it also provides the `MetaInfo` directly, no null checks required) (the placements are from the previous code before rebasing :') )